lib/parachain/collation_fetching.go

```diff
@@ -0,0 +1,86 @@
+package parachain
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// CollationFetchingRequest represents a request to retrieve
+// the advertised collation at the specified relay chain block.
+type CollationFetchingRequest struct {
+	// Relay parent we want a collation for
+	RelayParent common.Hash
+
+	// Parachain id of the collation
+	ParaID ParaID
+}
+
+// Encode returns the SCALE encoding of the CollationFetchingRequest
+func (c *CollationFetchingRequest) Encode() ([]byte, error) {
```

Comment:
Do we need to wrap marshal with an encode func? Typically I think we have avoided this unless the encoding in particularly complex/not scale

Reply:
I have implemented the 'network.Message' interface as 'CollationFetchingRequest' will be passed into [this](https://github.com/ChainSafe/gossamer/pull/3334/files#diff-5794e04e36068d98b7c6d2280c7e0b41fecca0b90bd6dd4576cd62b951ab587aR32) function as 'req'.